biscuit-auth/src/token/authorizer.rs

```diff
@@ -538,20 +543,25 @@ impl<'t> Authorizer<'t> {
     where
         error::Token: From<<R as TryInto<Rule>>::Error>,
     {
-        let rule = rule.try_into()?;
+        let rule = rule.try_into()?.convert(&mut self.symbols);
 
         self.world
             .run_with_limits(&self.symbols, limits.into())
             .map_err(error::Token::RunLimit)?;
-        let rule = rule.convert(&mut self.symbols);
-        let origin = if let Some(t) = self.token {
+        let all_origins = if let Some(t) = self.token {
             std::iter::repeat(())
                 .enumerate()
                 .take(t.block_count())
                 .map(|(i, _)| i)
                 .collect()
         } else {
-            [0].iter().collect()
+            [0, usize::MAX].iter().collect()
+        };
+
+        let origin = if rule.scopes.is_empty() {
+            all_origins
+        } else {
+            rule.origins(usize::MAX, Some(&self.public_key_to_block_id))
```

Comment:
if the rule has a `previous` scope then this will generate a list of scopes from 0 to usize::MAX

Reply:
We should disallow trusting previous altogether in the authorizer imo, it's too dangerous

Reply:
I agree we should remove it. If someone wants to use it anyway it's always possible to explicitely set the list of scopes